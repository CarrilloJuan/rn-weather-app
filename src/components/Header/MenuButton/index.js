import React from 'react';
import {useNavigation} from '@react-navigation/native';
import {TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/Entypo';

import Colors from '../../../constants/Colors';

export default function MenuButton() {
  const navigation = useNavigation();
  return (
    <TouchableOpacity
      onPress={() => navigation.toggleDrawer()}
      testID="menu-button">
      <Icon name="menu" size={30} color={Colors.dark} />
    </TouchableOpacity>
  );
}
