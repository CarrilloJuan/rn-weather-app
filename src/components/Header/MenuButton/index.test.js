/**
 * @jest-environment jsdom
 */

import React from 'react';
import {shallow, mount} from 'enzyme';
import MenuButton from '.';

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      toggleDrawer: mockedNavigate,
    }),
  };
});

describe('MenuButton Component', () => {
  const origConsole = console.error;
  beforeEach(() => {
    console.error = () => {};
  });
  afterEach(() => {
    console.error = origConsole;
    jest.clearAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('renders as expected', () => {
    const wrapper = shallow(<MenuButton />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call the toggleDrawer method', () => {
    const wrapper = mount(<MenuButton />);
    wrapper.find({testID: 'menu-button'}).at(0).props().onPress();
    expect(mockedNavigate.mock.calls.length).toBe(1);
  });
});
