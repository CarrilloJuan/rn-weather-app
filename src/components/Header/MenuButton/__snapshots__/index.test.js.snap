// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton Component renders as expected 1`] = `
<ForwardRef
  onPress={[Function]}
  testID="menu-button"
>
  <Icon
    allowFontScaling={false}
    color="#444"
    name="menu"
    size={30}
  />
</ForwardRef>
`;
